packages/typespec-ts/test/modularIntegration/generated/azure/core/src/api/operations.ts

```diff
@@ -260,12 +270,39 @@ export async function _listWithPageDeserialize(
 }
 
 /** List with Azure.Core.Page<>. */
-export async function listWithPage(
+export function listWithPage(
   context: Client,
   options: ListWithPageOptions = { requestOptions: {} }
-): Promise<Page> {
-  const result = await _listWithPageSend(context, options);
-  return _listWithPageDeserialize(result);
+): PagedAsyncIterableIterator<Page> {
+  return listWithPageGenerator(context, options);
```

Comment:
I don't get why do we need to have the logic in a different function unless you plan on calling that function elsewhere too.

Reply:
my understanding is this is to make it simpler for customization -- trying to avoid having large amounts of code in an operation method to allow for simple wrappers and input/output transformations